src/sentry/api/event_search.py

```diff
@@ -505,15 +505,24 @@ def key_mappings_lookup(self):
         return lookup
 
     def is_numeric_key(self, key):
-        return key in self.config.numeric_keys or is_measurement(key) or is_span_op_breakdown(key)
+        return (
+            key in self.config.numeric_keys
+            or is_measurement(key)
+            or is_span_op_breakdown(key)
+            or self.builder.get_field_type(key) == "number"
+        )
 
     def is_duration_key(self, key):
         return (
             key in self.config.duration_keys
             or is_duration_measurement(key)
             or is_span_op_breakdown(key)
+            or self.builder.get_field_type(key) == "duration"
```

Comment:
```suggestion
             or self.builder.get_field_type(key) in DURATION_UNITS
```

Reply:
I think get_field_type is returning the unit instead of the actual field type

Reply:
Might need both 
```
or self.builder.get_field_type(key) == "duration"
or self.builder.get_field_type(key) in DURATION_UNITS`
```
Not sure if ` self.builder.get_field_type(key) == "duration"` will be used outside of CM

Reply:
good catch!